.github/workflows/tests-evm.yml

```
@@ -0,0 +1,103 @@
+name: EVM test suite
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+  merge_group:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  preflight:
+    uses: ./.github/workflows/reusable-preflight.yml
+
+  evm-test-suite:
+    needs: [preflight]
+    runs-on: ${{ needs.preflight.outputs.RUNNER }}
+    if: ${{ needs.preflight.outputs.changes_rust }}
+    timeout-minutes: 60
+    container:
+      image: ${{ needs.preflight.outputs.IMAGE }}
+    env:
+      # Enable debug assertions since we are running optimized builds for testing
+      # but still want to have debug assertions.
+      RUSTFLAGS: "-C debug-assertions"
+      RUST_BACKTRACE: 1
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: script
+        run: |
+          forklift cargo build --locked --profile production -p pallet-revive-eth-rpc --bin eth-rpc
+          forklift cargo build --bin substrate-node
+
+      - name: Checkout evm-tests
+        uses: actions/checkout@v4
+        with:
+          repository: paritytech/evm-test-suite
+          ref: 6f95f703b1bc25d7eb2ca9377bd2c422e7a50ce9
+          path: evm-test-suite
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 22
+
+      - name: script
+        run: |
+          echo "Change to the evm-test-suite directory"
+          cd evm-test-suite
+          echo "Download the resolc binary"
+          wget https://github.com/paritytech/revive/releases/download/v0.1.0-dev.9/resolc -q
+          chmod +x resolc
+          mv resolc /usr/local/bin
+          resolc --version
+
+          echo "Check that binaries are in place"
+          export NODE_BIN_PATH=$(readlink -f ../target/debug/substrate-node)
+          export ETH_RPC_PATH=$(readlink -f ../target/production/eth-rpc)
+          export RESOLC_PATH=/usr/local/bin/resolc
+          echo $NODE_BIN_PATH $ETH_RPC_PATH $RESOLC_PATH
+
+          echo "Install npm dependencies"
+          npm install
+          # cat matter-labs-tests/hardhat.config.ts | grep batchSize
+
+          echo "Installing solc"
+          wget https://github.com/ethereum/solidity/releases/download/v0.8.28/solc-static-linux -q
+          chmod +x solc-static-linux
+          mv solc-static-linux /usr/local/bin/solc
+          echo "Run the tests"
+          # Tests don't work with batchSize 300 on self-hosted runners in docker container
+          sed -i 's/batchSize: 300,/batchSize: 100,/' matter-labs-tests/hardhat.config.ts
```

```suggestion
          export BATCH_SIZE=100
```
Now we can just define it as an env var

---

Thanks! Added it to step env variables.